docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,448 @@
-    "my_doubled_field": {
-      "script": {
-        "lang":   "expression",
-        "source": "doc['my_field'] * multiplier",
-        "params": {
-          "multiplier": 2
-        }
-      }
-    }
-  }
-}
--------------------------------------
-
-[discrete]
-=== Script parameters
 
 `lang`::
 
-    Specifies the language the script is written in.  Defaults to `painless`.
-
+    Specifies the language the script is written in. Defaults to `painless`.
 
 `source`, `id`::
 
-    Specifies the source of the script.  An `inline` script is specified
-    `source` as in the example above. A `stored` script is specified `id`
-    and is retrieved from the cluster state (see <<modules-scripting-stored-scripts,Stored Scripts>>).
-
+    Specifies the source of the script. An `inline` script is specified as `source` in the previous example. A `stored` script is specified by `id` and is retrieved from the cluster state using the `_scripts` endpoint. For
+    example, `_scripts/id`.
 
 `params`::
 
     Specifies any named parameters that are passed into the script as
-    variables.
-
-[IMPORTANT]
-[[prefer-params]]
-.Prefer parameters
-========================================
-
-The first time Elasticsearch sees a new script, it compiles it and stores the
-compiled version in a cache. Compilation can be a heavy process.
-
-If you need to pass variables into the script, you should pass them in as
-named `params` instead of hard-coding values into the script itself.  For
-example, if you want to be able to multiply a field value by different
-multipliers, don't hard-code the multiplier into the script:
-
-[source,js]
-----------------------
-  "source": "doc['my_field'] * 2"
-----------------------
-// NOTCONSOLE
-
-Instead, pass it in as a named parameter:
-
-[source,js]
-----------------------
-  "source": "doc['my_field'] * multiplier",
-  "params": {
-    "multiplier": 2
-  }
-----------------------
-// NOTCONSOLE
-
-The first version has to be recompiled every time the multiplier changes.  The
-second version is only compiled once.
-
-If you compile too many unique scripts within a small amount of time,
-Elasticsearch will reject the new dynamic scripts with a
-`circuit_breaking_exception` error. For most contexts, you can compile up to 75
-scripts per 5 minutes by default. For ingest contexts, the default script
-compilation rate is unlimited. You can change these settings dynamically by
-setting `script.context.$CONTEXT.max_compilations_rate` eg.
-`script.context.field.max_compilations_rate=100/10m`.
-
-========================================
+    variables. <<prefer-params,Use parameters>> instead of hard-coded values to decrease compile time.
 
 [discrete]
-[[modules-scripting-short-script-form]]
-=== Short script form
-A short script form can be used for brevity. In the short form, `script` is represented
-by a string instead of an object. This string contains the source of the script.
-
-Short form:
+[[hello-world-script]]
```

Comment:
I like the idea here, but in keeping with the practical nature of painless, let's keep the old example before.

What do you think about?
1) First show the end goal (assuming most people will copy and paste the first thing they see)
2) `return doc['my_field'] * 2;` // hard coded
3) `return doc['my_field'] * params['multiplier'];` // uses parameters
4) `doc['my_field'] * params['multiplier']` // removes semi-colon and return, explains how they are optional

2 & 3 can just provide the `source`, no need to provide the full request.  This reduces redundancy and prevents people from copy and pasting "bad" examples.

Reply:
I agree that this example is more practical. However, I was trying to introduce the script piecemeal, instead of injecting multiple concepts into a single example and then unwinding it.

Reply:
updated script to use `.value` and `"lang": "painless"`

Reply:
Great stuff @stu-elastic 🥳  I'm incorporating all of this goodness into the sections for how to write scripts.